apps/root/src/hooks/useServiceEvents.ts

```diff
@@ -2,21 +2,27 @@ import React from 'react';
 import { v4 as uuidv4 } from 'uuid';
 import { EventsManager } from '@services/eventsManager';
 
-function useServiceEvents<ServiceData, ServiceType extends EventsManager<ServiceData>>(
+function useServiceEvents<ServiceData, ServiceType extends EventsManager<ServiceData>, Key extends keyof ServiceType>(
   service: ServiceType,
-  property: keyof ServiceData
+  property: Key
 ) {
-  const [, setServiceData] = React.useState<ServiceData>(service.serviceData);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  type PropertyValue = ServiceType[Key] extends (...args: any[]) => any
+    ? ReturnType<ServiceType[Key]>
+    : ServiceType[Key];
+  const getServicePropertyData = () => (service[property] as () => PropertyValue).bind(service) as PropertyValue;
+
+  const [serviceData, setServiceData] = React.useState<PropertyValue>(getServicePropertyData());
 
   React.useEffect(() => {
     const callbackId = uuidv4();
-    service.setCallback(callbackId, setServiceData);
+    service.setCallback(callbackId, () => setServiceData(getServicePropertyData()));
```

Comment:
So in this case getServicePropertyData is actually a function that returns the function from the service, is this actually working?

Reply:
It works! But I understand your point, I'm thinking now that it's more clear if getServicePropertyData, gets the service property data instead of the function that does that 😄